base/rational.jl

```diff
@@ -468,21 +468,21 @@ for (S, T) in ((Rational, Integer), (Integer, Rational), (Rational, Rational))
     end
 end
 
-trunc(::Type{T}, x::Rational) where {T} = round(T, x, RoundToZero)
-floor(::Type{T}, x::Rational) where {T} = round(T, x, RoundDown)
-ceil(::Type{T}, x::Rational) where {T} = round(T, x, RoundUp)
-
 round(x::Rational, r::RoundingMode=RoundNearest) = round(typeof(x), x, r)
 
 function round(::Type{T}, x::Rational{Tr}, r::RoundingMode=RoundNearest) where {T,Tr}
-    if iszero(denominator(x)) && !(T <: Integer)
+    if iszero(denominator(x))
         return convert(T, copysign(unsafe_rational(one(Tr), zero(Tr)), numerator(x)))
     end
     convert(T, div(numerator(x), denominator(x), r))
 end
 
-function round(::Type{T}, x::Rational{Bool}, ::RoundingMode=RoundNearest) where T
-    if denominator(x) == false && (T <: Integer)
+function round(::Type{T}, x::Rational, r::RoundingMode=RoundNearest) where {T<:Integer}
+    convert(T, div(numerator(x), denominator(x), r))
+end
+
+function round(::Type{T}, x::Rational{Bool}, ::RoundingMode) where {T<:Integer}
+    if denominator(x) == false
```

Comment:
I like that you switched rationals to also use the generic definitions. That should make these changes unnecessary.